website/static/js/components/autocomplete.js

```diff
@@ -0,0 +1,189 @@
+            },
+            source: this.substringMatcher(this.items())
+        });
+
+        this.inputElement.bind('typeahead:selected', function(event, item) {
+            // Call the parent viewModel's onSelected
+            this.onSelected(item.value);
+        }.bind(this));
+    },
+    /** Clear the selection */
+    onClear: function () {
+        this.inputElement.typeahead('val', '');
+        this.itemSelected(null);
+    },
+    /** Set the item as selected */
+    onSelected: function (item) {
+        this.itemSelected(item);
+    }
+});
+
+
+var DraftRegistrationsSearchViewModel = function (params) {
+    BaseSearchViewModel.apply(this, arguments);
+};
+$.extend(DraftRegistrationsSearchViewModel.prototype, BaseSearchViewModel.prototype, {
+    processData: function (data) {
+        this.items(data.draftRegistrations);
+    },
+    onSubmit: function () {
+        // Redirect the user to continue the draft registration
+        window.location = this.itemSelected().url;
+    },
+    suggestionTemplate: function(item) {
+        var dateUpdated = new $osf.FormattableDate(item.value.dateUpdated);
+        var dateCreated = new $osf.FormattableDate(item.value.dateCreated);
+        // jQuery implicity escapes HTML
+        return $('<div>').append(
+            $('<p>', {
+                text: item.value.node.title
+            }).append(
+                [
+                    $('<p>').append(
+                        $('<small>', {
+                            className: 'm-l-md text-muted',
+                            text: 'Initiated by: ' + item.value.initiator.name
+                        })
+                    ),
+                    $('<p>').append(
+                        $('<small>', {
+                            className: 'm-l-md text-muted',
+                            text: 'Initiated: ' + dateCreated.local
+                        })
+                    ),
+                    $('<p>').append(
+                        $('<small>', {
+                            className: 'm-l-md text-muted',
+                            text: 'Last updated: ' + dateUpdated.local
+                        })
+                    )                    
+                ]
+            )                
+        ).html();
+    },
+    substringMatcher: function(strs) {
+        return function findMatches(q, cb) {
+            var matches = [];
+            var substrRegex = new RegExp(q, 'i');
+            var count = 0;
+
+            $.each(strs, function(i, str) {
+                if (substrRegex.test(str.node.title)) {
+                    count += 1;
+                    matches.push({ value: str });
+                }
+            });
+
+            cb(matches);
+        };
+    }
+});
+
+
+ko.components.register('osf-draft-registrations-search', {
+    viewModel: {
+        createViewModel: function(params, componentInfo) {
+            // Inject the inputElement
+            $.extend(params, {
```

Comment:
Minor: prefer to copy params objects rather than mutating them. Also, defaults should go *before* the overrides:

```js
var opts = $.extend({}, {inputElement: ...}, params};
 return new DraftRegistrationsSearchViewModel(opts);
```